WalletWasabi.Gui/Controls/NoparaPasswordBox.cs

```diff
@@ -401,11 +327,7 @@ private void PaintText()
 				_ = DisplayWarningAsync(PasswordHelper.TrimmedMessage);
 			}
 
-			Text = _displayText.Substring(0, password.Length);
-			if (IsPasswordVisible)
-			{
-				Text = password;
-			}
+			Text = IsPasswordVisible ? password : new string(ReplacementChar, password.Length);
```

Comment:
I think this is a good opportunity to improve this. Do not manipulate the `Text`, use a converter instead.

Reply:
Please dont do this, or use a converter (sorry Lucas :) ),

Simply set the PasswordChar property of the textbox!